hudi-client/hudi-spark-client/src/main/java/org/apache/hudi/keygen/SparkKeyGeneratorInterface.java

```diff
@@ -28,4 +30,6 @@
   String getRecordKey(Row row);
 
   String getPartitionPath(Row row);
+
+  String getPartitionPath(InternalRow internalRow, StructType structType);
```

Comment:
should we do a default implementation, so that not everyone is forced to implemment this?

Reply:
I followed what we did when we introduced Row to these interfaces. We added default impl only in BuiltInKeyGen.